File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxP = 0
        minS = prices[0]
        n = len(prices)
        for i in range(1, n):
            maxP = max(maxP, prices[i]-minS)
            minS = min(minS, prices[i])

        return maxP    
        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxP = 0, minS = prices[0], n = prices.size();
        for(int i=0; i<n; i++) {
            maxP = max(maxP, prices[i]-minS);
            minS = min(minS, prices[i]);
        }
        return maxP;
    }
};